pytheus/backends.py

```diff
@@ -126,16 +135,45 @@ def get(self) -> float:
 
 
 class MultipleProcessRedisBackend(Backend):
-    """Provides a multi-process backend that uses Redis."""
+    """
+    Provides a multi-process backend that uses Redis.
+    Single dimension metrics will be stored as list (ex. Counter) while labelled
+    metrics will be stored in an hash.
+    Currently is very naive, and there is a lot of room for improvement.
+    (ex. maybe store all dimensions in the same hash and be smart on retrieving everything...)
+    Note: currently not expiring keys
+    """
+
+    CONNECTION_POOL: redis.Redis = None
+
+    def __init__(self, config: BackendConfig, metric: 'Metric') -> None:
+        super().__init__(config, metric)
+        self._key_name = self.metric._collector.name
+        self._labels_hash = None
+        if self.metric._labels:
+            self._labels_hash = '-'.join(sorted(self.metric._labels.values()))
```

Comment:
used a `-` as separator, but no specific reasons for it. I think it would make it more readable when checking the db but at the same time if a label uses `-` it might get confusing but acceptable for now

Reply:
What about the space in memory?

Shall we just store a `sha256` built from the list of label values?

Reply:
I think we can open a ticket for a deeper discussion and analysis, by storing the sha it wouldn't be human readable anymore but probably better, but I would say outside the scope of this week

Reply:
Sounds good, my worry was that storing the label values has O(n) space complexity, whereas a hash is O(1).

If someone goes with huge cardinality they can crash memory or fill disk storage, but I agree we can leave this for later :+1:

Reply:
I'm not sure I understood 🤔 
You have to store a value entry for each possible labels value combination, whether it's the concatenation of the labels or an hash of those, they would still be the same number of entries 🤔